cache/client.go

```diff
@@ -35,47 +34,50 @@ type clientMetrics struct {
 	dataSize   *prometheus.HistogramVec
 }
 
-func newClientMetrics(reg prometheus.Registerer) *clientMetrics {
-	cm := &clientMetrics{}
-
-	cm.operations = promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+func newClientMetrics(regs []prometheus.Registerer) *clientMetrics {
```

Comment:
Idea to avoid changing the argument to a slice of Registerers here:

Would it work to implement a "tee registerer" in the following way and use it to bundle up all the registerer before passing them to `newClientMetrics`? This would probably also avoid the lint warning below.

```Go
type TeeRegisterer struct {
	Regs []prometheus.Registerer
}

func (t TeeRegisterer) Register(c prometheus.Collector) error {
	for _, reg := range t.Regs {
		if err := reg.Register(c); err != nil {
			return err
		}
	}
	return nil
}

func (t TeeRegisterer) MustRegister(cs ...prometheus.Collector) {
	for _, reg := range t.Regs {
		reg.MustRegister(cs...)
	}
}

func (t TeeRegisterer) Unregister(c prometheus.Collector) bool {
	result := false
	for _, reg := range t.Regs {
		if reg.Unregister(c) {
			result = true
		}
	}
	return result
}
```

Reply:
I agree with @beorn7 's idea. I think that should result in a much smaller diff as well. Can you give that a try @lamida?

Reply:
The PR is updated with that great suggestion. Now there is no more need to update the constructor.